crates/bevy_ecs/src/query/iter.rs

```diff
@@ -1089,13 +1089,20 @@ where
 
     #[inline(always)]
     fn next(&mut self) -> Option<Self::Item> {
-        let entity = self.entity_iter.next()?;
-        // SAFETY: `entity` is passed from `entity_iter` the first time.
-        unsafe { self.fetch_next(entity).into() }
+        while let Some(entity) = self.entity_iter.next() {
```

Comment:
If i understand correctly, this `while let Some()` isn't strictly necessary since the only public way of getting `QuerySortedIter` is via the various `QueryIter.sort()` methods, which provide an `entity_list` containing only items where `fetch` returns `Some(D::Item)` (and not `None`).

However it's still good to have as a precaution for internal code if we manually create a `QuerySortedIter`?